openedx/core/djangoapps/appsembler/eventtracking/utils.py

```
@@ -31,9 +31,16 @@ def get_site_config_for_event(event_props):
                 # allow to fail if more than one Organization to avoid sharing data
                 orgcourse = models.OrganizationCourse.objects.get(course_id=args)
                 org = orgcourse.organization
+            else:
+                raise exceptions.EventProcessingError(
```

@melvinsoft Why do we want to raise an exception here? Won't it just replace one exception message in the logs with another and not reduce the noise in the logs?

---

@johnbaldwin Your question makes sense,  but there is a try/except that wraps the call to this function which just drops the event when an `EventProcessingError` occurs, so it shouldn't block Figures.

---

@johnbaldwin Yes, the question makes sense, but this should stop the noise in the logs, since in the two places when it's been called, the exception is being handled. It actually makes sense to throw an exception if we cannot tie the event to an org.

The event will be sent to main Appsembler segment account, but not to the specific customer.

https://github.com/appsembler/edx-platform/blob/37fcb8eb805beb151d455057b77aee9c70daac63/openedx/core/djangoapps/appsembler/eventtracking/segment.py#L69-L77

and here

https://github.com/appsembler/edx-platform/blob/appsembler/tahoe/master/common/djangoapps/track/shim.py#L144-L161